### DISCUSSION TITLE
Library version number

### DISCUSSION THREAD

**Discussion by pramsey** [google/s2geometry#277] [Q&A] [answered]
Is there an expected way to extract the version number from the library? Like, if I have the header files and library files, is there a function to call or something to tell me what version it is? I'm building out an autoconf script and want to be able to warn/fail if the version is not recent enough to support the features I use.

**Comment by jmr** [comment, marked as answer]
AFAIK, there's no good way to do it once it's been installed. Can you check for existence of functions you need in the autoconf scripts?

**Reply by jmr** [reply]
You may be able to look for `libs2.so.0.10.0`.

**Reply by pramsey** [reply]
Unfortunately autoconf only has really nice simple symbol checking for C libraries (and extern C symbols) so while it could be done (write and link small program that has all needed functions) I'm just going to go with a "hope your install has what we need" policy for now. A little C function that emits the version would be handy if S2 becomes more widely packaged (more and more geospatial projects are starting to depend on it, so the next stage will be hoping for independent packaging).
